### ISSUE TITLE
Public Chat

### ISSUE THREAD

**Issue by paulnotfromstrapi** [PaulBratslavsky/nextjs-build-in-public#1]
Share your thoughts and ideas and recommendations for this project.

**Comment by paulnotfromstrapi** [comment]
Hello everyone

**Comment by PaulBratslavsky** [comment]
hello

**Comment by ztxone** [comment]
Need more data to work with )

**Comment by PaulBratslavsky** [comment]
@ztxone we are just starting out.  So still in planing phase.  Did you watch the stream.  We outlined some of the project basics  requirements.  

We want to build an Events management website.  We are still in planning phase.  Areas we are looking for help

**Design for Pages**
   - home page
   - register page
   - login page
   - events list page
   - events detail page
   - profile page
   
Once we have the basic designs we will structure the data.
   
I just added PR for typescript support to be able to use Strapi types in the Next JS frontend.
   
Next step is to design the Front page. We were thinking of keeping it simple.  Hero section.  List of upcoming events.  And in the hero section have a link to login/register page.

Once we have the home page, start working on Register Page and login in page.  We will keep it simple to start.

we will use email, username and password to register

and login will be done with an identifier ( email/username) and password.

We will receive a JWT token from Strapi so will have to figure out how to stroe it in session storage

In terms of how to handle form submission.  I know we can use client components.  But I was curios of possibly using server actions.  Still up for debate.

Let me know if this helps. Or any suggestion you may have.

**Comment by wongprom** [comment]
Hi all!
Just wanted to drop by and drop some ❤️ for this project and hopefully try to contribute to this project. 
I have a PR template that may be helpful for this project; you can, of course, tweak it as needed.

**PR Template**
| Status  | Type  | Env Vars Change | Review App | Ticket |
| :---: | :---: | :---: | :--: | :--: |
| Ready/Hold | Feature/Bug/Tooling/Refactor/Hotfix | Yes/No | [Link](<Review app link here>) | [Link](<ticket link here>) |

> ⚠️ NOTE: use notes like this to emphasize something about the PR. This could include other PRs this PR is built on top of; new or removed environment variables; reasons for why the PR is on hold; or anything else you would like to draw attention to.

## Problem

_What problem are you trying to solve?_


## Solution

_How did you solve the problem?_


## Before & After Screenshots

**BEFORE**:
[insert screenshot here]

**AFTER**:
[insert screenshot here]


## Other changes (e.g. bug fixes, UI tweaks, small refactors)


## Deploy Notes

_Notes regarding deployment of the contained body of work. These should note any
new dependencies, new scripts, etc._

**New environment variables**:

- `env var` : env var details

**New scripts**:

- `script` : script details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details


## Checklist



General

- [ ] My pull request represents one logical piece of work.
- [ ] My commits are related to the pull request and look clean.
- [ ] I have performed a self-review of my code
- [ ] Vercel deployments pass with my changes
- [ ] My code contains no conflicts and is up to date with the latest main branch

**Comment by ztxone** [comment]
> Let me know if this helps. Or any suggestion you may have.

Thank for details! 

How do you plan to sync starpi data between contributors?
technically now everyone have it's own instance of strapi-data on localhost...

**Comment by Boegie19** [comment]
we should probebly have an export of some test data in the repo itself so you can import it when you need to have data. only issue would if 2 people update the zip it would  be merge comflicts that can't be fixed.

**Comment by ztxone** [comment]
> we should probebly have an export of some test data in the repo itself so you can import it when you need to have data. only issue would if 2 people update the zip it would be merge comflicts that can't be fixed.

Yes, think so too. 
And one else problem:

after cloning repo i don't have any env variables and even after yarn & yarn build (in backend directory) i can't start backend...
so do i need to totally rebuild strapi from 0 for me
or it has some ability to copy same vars like APP_KEYS API_TOKEN_SALT etc to my localhost?

**Comment by ztxone** [comment]
one else thing 
after strapi export command - i guess strapi exports admin user too, isn't it? 
so when i will import exported data i will need same admin account to strapi admin panel?

**Comment by PaulBratslavsky** [comment]
@wongprom thanks for sharing the template.  I will add it.

**Comment by PaulBratslavsky** [comment]
I have added few todos in the issues, feel free to work on some.

**Comment by ztxone** [comment]
Made another PR. Correct me if i make something wrong about PRs... a bit newbie in that)

**Comment by PaulBratslavsky** [comment]
@ztxone no worries.  I ended up changing few things in your initial PR to make rendering images easier when pulling from Strapi.  But I ended up merging those changes late last night.  So I think I created conflicts in the Main branch vs your new PR.

I will look to update.  Also, have you ever done a rebase before?  

And also, I am fairly new to managing public projects with multiple contributors, so it's a learning lesson for me too.

**Comment by ztxone** [comment]
I see. There is a problem for me with permissions to your repo. So i was asked for fork you repo cuz i don't have rights to commit original branches. This case is new for my experience too )) 
usually my account adds to repo as a contributor, so i'm working with same repo but in my own branch. This way causes no problem with PR and merge.
At this point idk what to do with your commits on my fork and how it will be handled. Will try to do experiments)

**Comment by PaulBratslavsky** [comment]
@ztxone you can ignore, I ended up creating a branch with your changes and merging that instead.  So moving forward, just pull the new code to your main branch on you local computer and start from there.  

I made the updates here https://github.com/PaulBratslavsky/nextjs-build-in-public/pull/29

Also I will be hanging out in Discord https://discord.com/invite/strapi 12:30 CST - 1:50 PM CST in Strapi's Open Office Hour Chat, cant stop by if available, I can walk you through all the current changes.

**Comment by PaulBratslavsky** [comment]
@ztxone I again messed up. I realized what I was doing wrong.  I forgot fork your PR and fetch locally and make changes directly in your PR.  But I now know what I was doing wrong.  

I fixe the issue you were having.  There is a 'bug' in the type that you were using. 

I replaced it with a custom type.  But that is what was throwing you off. 

I pulled your changes into this branch.

Also just a note.  Every time you start a new branch, make sure you pull all the data from main branch so you are always working from the most recent code.

This is something I am learning to.

Your design looks great.  Awesome job as always.
![update](https://github.com/PaulBratslavsky/nextjs-build-in-public/assets/6153188/ffd9f2da-b948-42af-ab87-e751f5c39d34)

https://github.com/PaulBratslavsky/nextjs-build-in-public/pull/34

**Comment by ztxone** [comment]
Yep, need to refactor list-item a bit to my mind )

**Comment by PaulBratslavsky** [comment]
Lol you were doing everything right.  It was me pulling your PR to my main rather then fetching your remote repo. 

And having the code current with the new changes makes a big difference thank you.

I think this is the next good page to work on https://github.com/PaulBratslavsky/nextjs-build-in-public/issues/20
@ztxone you are doing a really good job.  I appreciate it.  Thank you.🙂

**Comment by ztxone** [comment]
> I think this is the next good page to work on #20

Done) I think it looks good!

**Comment by PaulBratslavsky** [comment]
@ztxone thank you for all your awesome help.  I am still trying to figure out how to add image via sever actions in next.  Before I can finish profile and add event pages.  

In terms of UI, you can work on this issue https://github.com/PaulBratslavsky/nextjs-build-in-public/issues/19

Let me know if you have any questions. 

And once again thank you for the help.

**Comment by PaulBratslavsky** [comment]
@ztxone i did add changes to the events page, but I am using the same card component.  But we can change it to the updated card.  I just wanted to make the fetch request to events page.

**Comment by PaulBratslavsky** [comment]
@ztxone I am slowly making progress: a bunch of Strapi side of things that need to be done.  I just updated the events page to pull the image, title, and description from strapi.  It now renders the Markdown found in the content field that I added. 

The markdown css found in `global.css` has the css responsible for the styling of the markdown.  Feel free to improve the styling.

![demo-markdown](https://github.com/PaulBratslavsky/nextjs-build-in-public/assets/6153188/6f36a5c1-6865-4811-a593-3acc08be957e)


When pulling the changes, you will need to run

``` bash
  yarn setup
  yarn import 
```

To install new packages and get the new seed data.

Feel free to work on it.  If you have other suggestions or UI improvements, feel free to share.

**Comment by PaulBratslavsky** [comment]
@ztxone new updates look awesome.   Thank you.

<img width="1350" alt="Screenshot 2023-11-12 at 1 45 08 PM" src="https://github.com/PaulBratslavsky/nextjs-build-in-public/assets/6153188/43ecc093-1355-4bb0-b382-68daf80b7811">

**Comment by PaulBratslavsky** [comment]
@ztxone completely random, would you ever want to come on my YT channel and teach me some vue js? 

In terms of this project.  I am almost done with the Strapi side of things on this app.  

Then, we want to test the functionality that we currently have, do final fixes, and deploy it as alpha for public users to play with.  

I also made some changes to the table and filtering; we now have one table, and you can use search to filter. I considered adding buttons to filter "DRAFT, PUBLIC, or ARCHIVED."

If you want ( have time ), you can do the final UI check for V1 and make any changes you feel make sense.

And a huge thank you for all your help.  You are awesome.

**Comment by ztxone** [comment]
> @ztxone completely random, would you ever want to come on my YT channel and teach me some vue js?

Yes, sure! Why not ) BTW we can try to build another frontend for this build-in-public project but with Nuxt3 ))) This showcase is good for HeadlessCMS.


> If you want ( have time ), you can do the final UI check for V1 and make any changes you feel make sense.

Yes, definitely i should see your last changes, cuz i have some commits in my fork, but not PR'ed yet. Last week was small free time actually( i'm sorry for that

**Comment by PaulBratslavsky** [comment]
I like the idea of rebuilding it with Vue JS.  That will be awesome—also, no need to apologize. I appreciate all the help you have already given.  

Also, I hope I did not break things too much for your PRs with all the changes I have made.😅.

**Comment by ztxone** [comment]
> Also, I hope I did not break things too much for your PRs with all the changes I have made.😅.

i'm always afraid of it when do PRs )))) While i'm working on my branch and there are a lot of another commits from other developers) I'm imaging how difficult is merging PR in this case!

**Comment by PaulBratslavsky** [comment]
Yeah, I have the same fear.  When you make the PR, I will rebase it with the main review manually and go from there.  Lol.  Part of it is my fault; I made lots of changes.  So I did it to myself.  But it will be fine.🙂
